tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/ApplyCondition.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Azure.Sdk.Tools.TestProxy.Common
+{
+    public class ApplyCondition
```

Comment:
nit: suggest renaming to just Condition as verbs are generally reserved for methods.

Reply:
I see where you're coming from here.

Originally was going to have a `MatchCondition` vs a `ApplyCondition`.  I'll revisit this.

Reply:
I like MatchCondition but I can see that being confusing with the concept of Matchers.

Reply:
We will settle on an answer here when I submit a PR with the `MatchCondition`. Two names enter, one name leaves.

(after I'm back in Jan 👍 )